src/network/network.rs

```diff
@@ -0,0 +1,162 @@
+
+    #[serde(rename = "name")]
+    pub name: String,
+
+    #[serde(rename = "network_interface")]
+    pub network_interface: Option<String>,
+
+    #[serde(rename = "options")]
+    pub options: Option<HashMap<String, String>>,
+
+    #[serde(rename = "subnets")]
+    pub subnets: Option<Vec<Subnet>>,
+}
+
+// NetworkOptions for a given container.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NetworkOptions {
+    #[serde(rename = "container_id")]
+    pub container_id: String,
+
+    #[serde(rename = "container_name")]
+    pub container_name: String,
+
+    #[serde(rename = "networks")]
+    pub networks: HashMap<String, PerNetworkOptions>,
+
+    #[serde(rename = "port_mappings")]
+    pub port_mappings: Option<Vec<PortMapping>>,
+}
+
+// PerNetworkOptions are options which should be set on a per network basis
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PerNetworkOptions {
+    #[serde(rename = "aliases")]
+    pub aliases: Option<Vec<String>>,
+
+    #[serde(rename = "interface_name")]
+    pub interface_name: String,
+
+    #[serde(rename = "static_ips")]
+    pub static_ips: Option<Vec<String>>,
+
+    #[serde(rename = "static_mac")]
+    pub static_mac: Option<String>,
+}
+
+// PortMapping is one or more ports that will be mapped into the container.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PortMapping {
+    #[serde(rename = "container_port")]
+    pub container_port: u16,
+
+    #[serde(rename = "host_ip")]
+    pub host_ip: String,
+
+    #[serde(rename = "host_port")]
+    pub host_port: u16,
+
+    #[serde(rename = "protocol")]
+    pub protocol: String,
+
+    #[serde(rename = "range")]
+    pub range: u16,
+}
+
+// StatusBlock contains the network information about a container
+// connected to one Network.
+#[derive(Serialize, Deserialize)]
+pub struct StatusBlock {
+    #[serde(rename = "dns_search_domains")]
+    pub dns_search_domains: Option<Vec<String>>,
+
+    #[serde(rename = "dns_server_ips")]
+    pub dns_server_ips: Option<Vec<String>>,
+
+    #[serde(rename = "interfaces")]
+    pub interfaces: Option<HashMap<String, NetInterface>>,
+}
+
+// NetInterface contains the settings for a given network interface.
+#[derive(Serialize, Deserialize)]
+pub struct NetInterface {
+    #[serde(rename = "mac_address")]
+    pub mac_address: String,
+
+    #[serde(rename = "networks")]
+    pub networks: Option<Vec<NetAddress>>,
+}
+
+// NetAddress contains the subnet and gatway.
+#[derive(Serialize, Deserialize)]
+pub struct NetAddress {
+    #[serde(rename = "gateway")]
+    pub gateway: Option<String>,
+
+    #[serde(rename = "subnet")]
+    pub subnet: IpNet,
+}
+
+// IPNet is used as custom net.IPNet type to add Marshal/Unmarshal methods.
+// TODO: [Not important] Transformed from golang net.IPNet. Swtich to native rust if needed.
+#[derive(Serialize, Deserialize)]
+pub struct IpNet {
+    #[serde(rename = "IP")]
+    pub ip: String,
+
+    #[serde(rename = "Mask")]
+    pub mask: String,
+}
```

Comment:
Golang should unmarshal this into a string with the format `IP/CIDIR` e.g to `192.168.1.1/24` or `fd80::1/64` for ipv6.
Can rust read this correctly? Is there a better type than string for a ip address?

Reply:
@Luap99 I have specially mentioned this in a TODO line just above this struct , yes as of now this will work without any problem. We can switch a rust native type later or cast it to a native type during business logic.

Reply:
So parsing for json should not be a problem at all.

Reply:
Edit: @Luap99 i think we should store IpNet as a single string of `IP/CIDR` and do the parsing ourselves. I'll edit this type.

Reply:
@Luap99 Added a new commit https://github.com/containers/netavark/pull/5/commits/07888fb3783aee2a444cfa8074ac1a2ca1723ad0 we are not using `IpNet` as of now instead accepting Subnet as string and we can cast it to `IpNet`. Will add automatic sterilizer for this later as an improvement.